src/init.luau

```diff
@@ -125,7 +145,9 @@ function encode(value: any, state: EncodeState, isStatic: boolean?)
 	elseif typeof(value) == "Vector3" then
 		return string.pack("<nnn", value.X, value.Y, value.Z), typeIndex
 	elseif typeof(value) == "CFrame" then
-		return string.pack(`<{string.rep("nnn", 4)}`, value:GetComponents()), typeIndex
+		local caseId = table.find(CFRAME_SPECIAL_CASES, value.Rotation)
```

Comment:
Right now `caseId` doesn't really do anything beneficial, since below the CFrame's full components list is still saved anyways, so I am not entirely sure I understand what the point of it is. I am making a guess that you originally meant to conditionally save the components.

This should save only the `caseId` byte if the CFrame is in the special cases list and three numbers for the position. If it isn't (`caseId` 0) the entire CFrame's components should get saved with the byte as just `0`.

This seems to already be accounted for in the code below.

Reply:
The `packFormat` variable (defined as `string.rep("nnn", if caseId == nil then 4 else 1)`) would be "nnnnnnnnnnnn" if `caseId` is nil (meaning it is not a special CFrame case), whereas it would be "nnn" if `caseId` *isn't* nil, which means it would only store the position, and not the rotation matrix.

The reason I still store `caseId`, even if the full list of components is still present, is because I am unsure on how I would identify that it is a special CFrame case when decoding.